frontend/tests/utils/downloadUtils.ts

```diff
@@ -0,0 +1,357 @@
+      break;
+    case "disease-filter":
+      diseaseFilterValue = filterText || noSelectionText;
+      break;
+    case "self-reported-ethnicity-filter":
+      ethnicityFilterValue = filterText || noSelectionText;
+      break;
+    case "sex-filter":
+      sexFilterValue = filterText || noSelectionText;
+      break;
+    case "no-filter":
+      break;
+    default:
+      throw new Error(`Invalid filter name: ${filterValue}`);
+  }
+
+  expect(
+    data[3].includes(`# Dataset Filter Values: ${datasetFilterValue}`)
+  ).toBeTruthy();
+  expect(
+    data[4].includes(`# Disease Filter Values: ${diseaseFilterValue}`)
+  ).toBeTruthy();
+  expect(
+    data[5].includes(
+      `# Self-Reported Ethnicity Filter Values: ${ethnicityFilterValue}`
+    )
+  ).toBeTruthy();
+  expect(
+    data[6].includes(`# Sex Filter Values: ${sexFilterValue}`)
+  ).toBeTruthy();
+};
+
+export async function downloadGeneFile(
+  page: Page,
+  tissues: string[],
+  subDirectory: string,
+  fileTypes: string[] = ["png"]
+): Promise<void> {
+  const allFileTypes = ["csv", "png", "svg"];
+  const dirPath = `${downLoadPath}/${subDirectory}`;
+
+  //wait for download file
+  const downloadPromise = page.waitForEvent("download");
+
+  //click the download icon
+  await page.getByTestId("download-button").click();
+  const CHECK = "Mui-checked";
+
+  for (const ext of allFileTypes) {
+    const checkboxId = `${ext}-checkbox`;
+    // uncheck if file type is checked but not
+    if (
+      (await page.getByTestId(checkboxId).getAttribute("class"))?.includes(
+        CHECK
+      ) &&
+      !fileTypes.includes(ext)
+    ) {
+      await page.getByTestId(checkboxId).locator("input").click();
+    }
+
+    // ensure wanted file types are checked
+    if (
+      !(await page.getByTestId(checkboxId).getAttribute("class"))?.includes(
+        CHECK
+      ) &&
+      fileTypes.includes(ext)
+    ) {
+      await page.getByTestId(checkboxId).locator("input").click();
+    }
+  }
+  await page.getByTestId("dialog-download-button").click();
+  const download = await downloadPromise;
+
+  // download can be zipped or not depending on number of tissues
+  let fileName = `${dirPath}/download.zip`;
+  if (fileTypes.length === 1 && tissues.length === 1 && fileTypes[0] == "svg") {
+    fileName = `${dirPath}/${tissues[0]}.${fileTypes[0]}`;
+  }
+
+  await download.saveAs(fileName);
+  //extract zip file
+  if (fileName.includes("zip")) {
+    const zip = new AdmZip(fileName);
+    zip.extractAllTo(dirPath);
+  }
+}
+
+export const getFilterText = async (page: Page, filterName: string) => {
+  const filter_label = `${getTestID(filterName)} [role="button"]`;
+  return await page.locator(filter_label).textContent();
+};
+
+export function verifyPng(
+  dirPath: string,
+  tissues: string[],
+  subDirectory: string
+) {
+  tissues.forEach(async (tissue) => {
+    // Capture the actual screenshot and compare it with the expected screenshot
```

Comment:
I see that SVG test is able to just compare the heatmap on the webpage vs. the downloaded heatmap SVG, why can't we do the same with PNG? Thank you!

Reply:
@victorikeokoro6 pls address this feedback.

Reply:
@tihuan the reason we cannot do the same is because the downloaded PNG has extra header texts which we don't have on the web. With SVG we were lucky because we load the SVG into the browser and take the screenshot of the part we want to compare. To do the same with PNG we would need to crop the image, which raise lots of accuracy issues.

Reply:
Ah got it! Thanks for the context, @charles-testco 🙏 

Since dev/staging/prod all have different data, I'm thinking we might need to go with the cropping image route instead of comparing the browser heatmap with the saved snapshot, otherwise the test will just fail at every deployment stage 🤔 

In terms of cropping the PNG image, I'm hoping that the offset X/Y for the heatmap portion stays the same/won't change frequently, so hopefully we can just hard code where we'll start cropping the image to isolate the heatmap portion out for diffing?

What do you think?

Thanks so much! Image diffing is so hard 😆

Reply:
Hey @charles-testco any thoughts on this :smile: ? Thank you!

Reply:
oh sorry @tihuan just seen this. Let me think about and discuss with Victor how best to tackle this?

Reply:
The issues have been resolved